search_action_domain/tests/test_models.py

```diff
@@ -0,0 +1,42 @@
+from odoo.tests import Form, TransactionCase, tagged
+from odoo.tools.safe_eval import safe_eval
+
+
+@tagged("models")
+class TestModels(TransactionCase):
+    def setUp(self):
+        super().setUp()
+        self.partner_model = self.env.ref("base.model_res_partner")
+
+    def create_server_action(self, code=""):
+        search_action = Form(self.env["ir.actions.server"])
+        search_action.name = "TEST: test"
+        search_action.model_id = self.partner_model
+        search_action.state = "code"
+        search_action.code = code
+        search_action = search_action.save()
+        return search_action
+
+    def test_01_search_action_domain(self):
+        # Create the server action
+        search_action = self.create_server_action(
+            code="""
+elements = env['res.partner'].search([('is_company', '=', True)])
+action = {'domain': [('id', 'in', elements.ids)]}
+            """
+        )
+        # Execute "Create As Filter" method from the created server action
+        search_action.create_as_ir_filter()
+        # Check if the filter was created successfully
+        created_filter = self.env.ref(
+            "search_action_domain.ir_filters_server_action_%s" % search_action.id
+        )
+        self.assertTrue(created_filter)
```

Comment:
use `raise_if_not_found=False` in previous line, otherwise this one makes no sense, as it won't be reached even if filter is not found